drivers/sensor/lis2mdl/lis2mdl.c

```diff
@@ -174,21 +182,83 @@ static int lis2mdl_attr_set(const struct device *dev,
 	return 0;
 }
 
-static int lis2mdl_sample_fetch_mag(const struct device *dev)
+static int get_single_mode_raw_data(const struct device *dev,
+				    int16_t *raw_mag)
 {
 	struct lis2mdl_data *lis2mdl = dev->data;
-	int16_t raw_mag[3];
+	int rc = 0;
 
-	/* fetch raw data sample */
-	if (lis2mdl_magnetic_raw_get(lis2mdl->ctx, raw_mag) < 0) {
-		LOG_DBG("Failed to read sample");
+	rc = lis2mdl_operating_mode_set(lis2mdl->ctx, LIS2MDL_SINGLE_TRIGGER);
```

Comment:
I would do that in init() function, based on single_mode is on or not. Does it make sense to you?

Reply:
I tested it. It does not work in the init function. It should be called every time we want to read in the single-mode.

Reply:
Ah, yes. It seems legit!